ios/example/app.js

```diff
@@ -1,90 +1,175 @@
+  top: 100,
+  title: 'Scan and send mifare connect command'
 });
 
+startSearchBtn.addEventListener('click', function() {
+  if (!nfcAdapter.isEnabled(sessionType)) {
+  Ti.API.error('This device does not support NFC capabilities!');
+  return;
+}
+nfcAdapter.begin(sessionType); // This is required for iOS only. Use "invalidate()" to invalidate a session.
+});
 nfcAdapter.addEventListener('didDetectTags', function (e) {
-    var mifare = nfcAdapter.createTagTechMifareUltralight({'tag':e.tags[0]});
-    mifare.addEventListener('didConnectTag', function (e) {
-        Ti.API.info('Connected tag object : ' + e.tag);
-        alert('Tag Connected: ' + e.tag);
-    });
-    mifare.connect({mifare});
-    nfcAdapter.invalidate(sessionType);
+    if (e.errorCode){
+        Ti.API.info('Error when detecting tag with - error code ' + e.errorCode + ' error domain: ' + e.errorDomain + ' error description ' + e.errorDescription);
+        logs.push('Error when detecting tag with - error code ' + e.errorCode + ' error domain: ' + e.errorDomain + ' error description ' + e.errorDescription);
+        setData(logs);
+    }else{
+        logs.push('Tag scanning start');
+        logs.push('Mifare tag detected');
+        setData(logs);
+    }
+  var mifare = nfcAdapter.createTagTechMifareUltralight({'tag':e.tags[0]});
+  mifare.addEventListener('didConnectTag', function (e) {
+      if (e.errorCode){
+          Ti.API.info('Error when connectiong tag with - error code ' + e.errorCode + ' error domain: ' + e.errorDomain + ' error description ' + e.errorDescription);
+          logs.push('Error when connectiong tag with - error code ' + e.errorCode + ' error domain: ' + e.errorDomain + ' error description ' + e.errorDescription);
+          setData(logs);
+      }else{
+          Ti.API.info('Connected tag object : ' + e.tag);
+          logs.push('Mifare tag connected : '  + e.tag);
+          setData(logs);
+      }
+      var sendMiFareCommand = mifare.sendMiFareCommand({data: [0x30,0x04]})
+      mifare.addEventListener('didSendMiFareCommand', function (e) {
+          if (e.errorCode){
+              Ti.API.info('Error when sending command with - error code ' + e.errorCode + ' error domain: ' + e.errorDomain + ' error description ' + e.errorDescription);
+              logs.push('Error when sending command with - error code ' + e.errorCode + ' error domain: ' + e.errorDomain + ' error description ' + e.errorDescription);
+              setData(logs);
+          }else {
+              Ti.API.info('Connected tag object : ' + e.responseDataLength);
+              logs.push('Command send to Mifare tag to read page data');
+              logs.push('Length of response object : ' + e.responseDataLength);
+              setData(logs);
+          }
+      });
+      nfcAdapter.invalidate(sessionType);
+  });
+   if (mifare){
+  mifare.connect({mifare});
+   }else {
+       alert('No mifare tag available to disconnect');
```

Comment:
i think you mean connect 
```suggestion
       alert('No mifare tag available to connect');
```

Reply:
Done